presto-main/src/main/java/com/facebook/presto/server/HttpRemoteTask.java

```diff
@@ -301,17 +309,15 @@ public synchronized void setOutputBuffers(OutputBuffers newOutputBuffers)
     }
 
     @Override
-    public int getPartitionedSplitCount()
+    public synchronized int getPartitionedSplitCount()
     {
-        int splitCount = pendingSourceSplitCount;
-        return splitCount + taskInfo.get().getStats().getQueuedPartitionedDrivers() + taskInfo.get().getStats().getRunningPartitionedDrivers();
+        return pendingSourceSplitCount + taskInfo.get().getStats().getQueuedPartitionedDrivers() + taskInfo.get().getStats().getRunningPartitionedDrivers();
     }
 
     @Override
-    public int getQueuedPartitionedSplitCount()
+    public synchronized int getQueuedPartitionedSplitCount()
```

Comment:
You are right. I did not see that StateMachine.state is volatile. I'll remove it.